### DISCUSSION TITLE
Helm and S3 Plugin

### DISCUSSION THREAD

**Discussion by kferrone** [verdaccio/charts#84] [Q&A]
Do I need to create my own image to include the s3 plugin. I can't seem to find anywhere that confirms or denies the existence of plugins within the default image used by the chart. I am planning on using v5 which seems to have the plugins just in the main source code. Does this mean the s3 plugin is just available?

**Comment by GregSharpe1** [comment]
@kferrone yes this is what I've done when working with S3 in aws. Here's an example of my Dockerfile:
```FROM verdaccio:5.10

USER root

RUN npm install --global-style --no-bin-links --no-optional verdaccio-aws-s3-storage && \
  cp -R ./node_modules/verdaccio-aws-s3-storage /verdaccio/plugins

USER 10001```

I don't recommend running the user as root.

**Reply by edclement** [reply]
Might be helpful too: example Dockerfile from the `verdaccio-prometheus-middleware` plugin I maintain: https://github.com/xlts-dev/verdaccio-prometheus-middleware#installation

**Comment by kferrone** [comment]
I see. Kind of a bummer that I have to have a one off custom image. It's not a big deal, it's just I have a bunch of control plane services and out of all of them, just one needed a custom image. 

I have an idea though, how about the Chart has a section in values.yaml like:
```yaml
plugins:
- s3
- something
- ldap
```

Then the main pod could use an init container to npm install the plugins into shared volume. Then the main container can add that node_modules to the `NODE_PATH` with an ENV var. 

Theoretically this would allow users to select plugins to install at runtime instead of requiring bespoke images.

**Reply by bjoerndanz** [reply]
I'm probably going to clone the chart to do exactly that
